lib/jormungandr/src/Cardano/Wallet/Jormungandr.hs

```diff
@@ -137,6 +142,21 @@ serveWallet (cfg, sb, tr) databaseDir listen lj beforeMainLoop = do
             let apiServer = Server.start settings tracerApi socket wallet
             race_ ipcServer apiServer
 
+    -- Will be used to connect "Cardano.Wallet.StakePool.Metrics" with
+    -- our networkLayer.
+    _toSPBlock :: Binary.Block -> (BlockHeader, PoolId)
+    _toSPBlock b =
+        (convertHeader header, fromJust $ Binary.producedBy header)
```

Comment:
Should definitely do something better here than `fromJust`…

Reply:
I added an error-message.

But this still doesn't feel right to me.

The PoolId might not be there because we're
1. looking at block0
2. on a BFT blockchain

For case 2) we'd probably crash anyway. Case 1) is actually pretty bad since we suddenly cannot retrieve the genesis block with a `NetworkLayer t (BlockHeader, PoolId)`.

I think I'll make `StakePool.Metrics` accept a `Maybe block`.

The only downside is that **if** something went wrong in the `Binary` module regarding poolIds, this would be silently swallowed.